app/jobs/import_job.rb

```diff
@@ -0,0 +1,65 @@
+class ImportJob < ActiveJob::Base
+  queue_as :default
+  require 'roo'
+
+  def perform(files_detail, klass)
+    files_detail.each do |klass, detail| 
+      import(detail[:path], detail[:name], klass.to_s.capitalize.constantize)
+    end
+  end
+  
+  def import(file_path, file_name, klass)
+    spreadsheet = open_spreadsheet(file_path, file_name)
+    header = spreadsheet.row(1)
+    if header == klass.new.attributes.keys
+      (2..spreadsheet.last_row).each do |i|
+        row = Hash[[header, spreadsheet.row(i)].transpose]
+        object = klass.find_by_id(row["id"]) if row["id"].present?
+        if row["id"].present? and object.present? 
+          #Skipping TODO: note id of skipped record
+        else
+          send("#{klass.class_name.downcase}_import", row.to_hash)
+        end
+      end
+    else
+      #notes The import CSV file was incorrectly formatted.
+    end  
+  end
+
+  def user_import(obj_hash)
+    object = User.new
+    object.attributes = obj_hash
+    object.password = User.create_password
+    object.sign_in_count = 0
+    object.save!
+  end
```

Comment:
Regarding your comment about errors, we probably should add some kind of rescue or error handling here.  On one hand this could mean one or more of:
- storing the full hash into a text_field for inspection
- attempting to identify the specific problem and recording it (not sure how feasible this is with mass assignment)
- just report a count of the number of rows that failed.
- other ideas?